sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -484,6 +488,59 @@ case class GpuCast(
     }
   }
 
+  private def castStructToString(input: GpuColumnVector): ColumnVector = {
+    var separatorColumn: ColumnVector = null
+    var spaceColumn: ColumnVector = null
+    val columns: ArrayBuffer[ColumnVector] = new ArrayBuffer[ColumnVector]()
+    val coreColumns: ArrayBuffer[ColumnVector] = new ArrayBuffer[ColumnVector]()
+
+    try {
+      withResource(GpuScalar.from(",", StringType)) { separatorScalar =>
+        separatorColumn = ColumnVector.fromScalar(separatorScalar, input.getRowCount().toInt)
+      }
+      withResource(GpuScalar.from(" ", StringType)) { separatorScalar =>
+        spaceColumn = ColumnVector.fromScalar(separatorScalar, input.getRowCount().toInt)
+      }
+      withResource(GpuScalar.from("[", StringType)) { bracketScalar =>
+        columns += ColumnVector.fromScalar(bracketScalar, input.getRowCount().toInt)
+      }
+      withResource(input.getBase().getChildColumnView(0)) { childView =>
+        columns += childView.asStrings()
+        coreColumns += childView.asStrings()
+      }
+
+      for(childIndex <- 1 until input.getBase().getNumChildren()) {
+        withResource(input.getBase().getChildColumnView(childIndex)) { childView =>
+          columns += separatorColumn
+          columns += spaceColumn.mergeAndSetValidity(BinaryOp.BITWISE_AND, childView)
```

Comment:
null handling of child columns is configurable in the latest versions of spark just like the brackets are, this has changed since 3.0.0. If it is not the legacy behavior, then we append "null" for null values.

 https://github.com/apache/spark/blob/4a3200b08ac3e7733b5a3dc7271d35e6872c5967/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L387

https://github.com/apache/spark/blob/4a3200b08ac3e7733b5a3dc7271d35e6872c5967/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L395

Otherwise it appears to be an empty string.

It would also be nice to have a comment explaining what is happening, because the BITWISE_AND is not obvious.

Reply:
Added some comments and there's a whole bunch of changes to support the new `3.1.0+` behavior with legacy support